File: BubblerApp/BubblerApp/Models/BlockedUser.swift
import Foundation

/// A user the current account has blocked (`GET /user/me/blocks`).
struct BlockedUser: Codable, Identifiable, Equatable {
    let id: Int
    let username: String
    let blockedAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case username
        case blockedAt = "blocked_at"
    }
}

File: BubblerAndroid/app/src/main/java/com/bubbler/android/data/model/BlockedUser.kt
package com.bubbler.android.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.Instant

/** A user the current account has blocked (`GET /user/me/blocks`). */
@Serializable
data class BlockedUser(
    val id: Int,
    val username: String,
    @SerialName("blocked_at")
    @Serializable(with = InstantIsoSerializer::class)
    val blockedAt: Instant,
)
